### DISCUSSION TITLE
In the 6.6.1-php8.3-apache image, where does the redirect from /wp-admin to /wp-admin/ occur?

### DISCUSSION THREAD

**Discussion by alexklibisz** [docker-library/wordpress#925] [General]
I'm running the 6.6.1-php8.3-apache in AWS ECS, being accessed via Cloudfront, pointed at API Gateway, pointed at an ECS container. In other words, it's basically two reverse proxies. Cloudfront proxies to a public URL on API Gateway. API Gateway proxies to an internal URL in ECS.

I'm finding that in some cases, requests to the cloudfront URL are being redirected to the API Gateway URL. One case in particular is when I request `<Cloudfront URL>/wp-admin`, I get a 301 redirect with an HTML body to `<API Gateway URL>/wp-admin/`.

The response and body look like this, with URLs redacted:

```
curl --verbose https://<Cloudfront URL>/wp-admin
...
< HTTP/2 301
< content-type: text/html; charset=iso-8859-1
< content-length: 383
< location: http://<API Gateway URL/wp-admin/
< date: Mon, 14 Oct 2024 05:05:07 GMT
< server: Apache/2.4.62 (Debian)
< apigw-requestid: fn70iia6IAMEahQ=
< x-cache: Miss from cloudfront
< via: 1.1 366f8b73c917b9815b93f1cf0b9f8e96.cloudfront.net (CloudFront)
< x-amz-cf-pop: SFO20-C1
< x-amz-cf-id: f1XqlqoXx34W8ZOoscsCXywyVfHBSHJzmRTMgY9TLMyDK2NUWqE3YA==
<
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="http://<API Gateway URL>/wp-admin/">here</a>.</p>
<hr>
<address>Apache/2.4.62 (Debian) Server at <API Gateway URL> Port 80</address>
</body></html>
```

AFAICT, this request never makes it into the `<VirtualHost:80>` configuration with `DocumentRoot /var/www/html`. I can tell this is the case because I added a header to that configuration: `Header set X-DEBUG "yes"`. When I request a non-redirected URL, I get the header back in the response. But when I request the `/wp-admin` URL and get a 301, I don't get the header back in the response. 

So I'm just stumped on how and where Apache is deciding that it should return the 301. I figure it's in some config file somewhere, but I've grepped around for 301, Redirect, Rewrite, etc., and haven't been able to find it.

I appreciate any tips!

**Comment by tianon** [comment]
I _think_ that's probably `mod_dir`, but I admit I haven't looked much into it: https://httpd.apache.org/docs/2.4/mod/mod_dir.html

**Comment by alexklibisz** [comment]
Thanks for the quick response. I looked a bit into that module, but I couldn't find anything obvious. But I think it's working now, after making two changes:

1. I updated API Gateway so that it passes the Host header set to my public URL to the Wordpress container.
2. I overrode the /etc/apache2/sites-available/000-default.conf file with a custom file that sets the ServerName to the public URL, injected as an env var.

My conf file looks like this:

```
<VirtualHost *:80>
    ServerName ${APACHE_SERVER_NAME}
    DocumentRoot /var/www/html
    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>
```
